openedx/core/djangoapps/config_model_utils/models.py

```diff
@@ -0,0 +1,125 @@
+
+    @classmethod
+    def current(cls, site=None, org=None, course=None):  # disable: pylint=arguments-differ
+        """
+        Return the current overridden configuration at the specified level.
+
+        Only one level may be specified at a time. Specifying multiple levels
+        will result in a ValueError.
+
+        Arguments:
+            site: The Site to check current values for
+            org: The org to check current values for
+            course: The course to check current values for
+
+        Returns:
+            An instance of :class:`cls.attribute_tuple()` with the overridden values
+            specified down to the level of the supplied argument (or global values if
+            no arguments are supplied).
+        """
+        if len([arg for arg in [site, org, course] if arg is not None]) > 1:
+            raise ValueError("Only one of site, org, and course can be specified")
+
+        if org is None and course is not None:
+            org = cls._org_from_course(course)
+
+        if site is None and org is not None:
+            site = cls._site_from_org(org)
+
+        stackable_fields = [cls._meta.get_field(field_name) for field_name in cls.STACKABLE_FIELDS]
+        field_defaults = {
+            field.name: field.get_default()
+            for field in stackable_fields
+        }
+
+        values = field_defaults.copy()
+
+        global_current = super(StackedConfigurationModel, cls).current(None, None, None)
+        for field in stackable_fields:
+            values[field.name] = field.value_from_object(global_current)
+
+        if site is not None:
+            site_current = super(StackedConfigurationModel, cls).current(site, None, None)
+            for field in stackable_fields:
+                value = field.value_from_object(site_current)
+                if value != field_defaults[field.name]:
+                    values[field.name] = value
+
+        if org is not None:
+            org_current = super(StackedConfigurationModel, cls).current(None, org, None)
+            for field in stackable_fields:
+                value = field.value_from_object(org_current)
+                if value != field_defaults[field.name]:
+                    values[field.name] = value
+
+        if course is not None:
+            course_current = super(StackedConfigurationModel, cls).current(None, None, course)
+            for field in stackable_fields:
+                value = field.value_from_object(course_current)
+                if value != field_defaults[field.name]:
+                    values[field.name] = value
+
+        return cls.attribute_tuple()(**values)
+
+    @classmethod
+    def _org_from_course(cls, course_key):
+        return course_key.org
+
+    @classmethod
+    def _site_from_org(cls, org):
+        configuration = SiteConfiguration.get_configuration_for_org(org)
+        if configuration is None:
+            return Site.objects.get(id=settings.SITE_ID)
```

Comment:
Do we need to handle the case where this query throws an error?

Reply:
Yes.

Reply:
Actually, I'm not sure there's any good way to handle that error, it should probably just throw.